### ISSUE TITLE
`modernize-use-emplace` not reported for `std::stack`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#56996]
```cpp
#include <stack>
#include <string>

struct A {
    std::string s;
};

void f1() {
    std::stack<A> s;
    s.push(A()); // no warning
}

void f2() {
    std::stack<A> s;
    s.push(A{}); // no warning
}

void f3() {
    std::stack<A> s;
    s.push(A{""}); // no warning
}

void f4() {
    std::stack<A> s;
    s.push({}); // no warning
}

void f5() {
    std::stack<A> s;
    s.push({""}); // no warning
}

void f6() {
    std::stack<A> s;
    s.emplace(A()); // warning
}
```
https://godbolt.org/z/sq6K8MchT

Interestingly the warning about the usage of unnecessary temporary objects is being reported so it seems like `std::stack` is not excluded from this check in general:
```console
<source>:35:15: warning: unnecessary temporary object created while calling emplace [modernize-use-emplace]]
    s.emplace(A()); // warning
              ^~~
```

**Comment by joeywatts** [comment]
I'd like to give this one a try.

**Comment by firewave** [comment]
I just saw that the documentation doesn't list this as a supported container. Sorry about not checking in advance. But that check has been around so long already that I just assumed it has the (complete) coverage.
